File: ios/NewsStand/Net/Repo/NewsRepository.swift
//
//  NewsRepository.swift
//  NewsStand
//
//  Created by Fatih Şen on 29.06.2019.
//  Copyright © 2019 Fatih Şen. All rights reserved.
//

import Foundation
import RxSwift

protocol NewsRepository {
	func addBookmark(news: News) -> Observable<News>
	func removeBookmark(news: News) -> Observable<News>
	func hasBookmark(url: String) -> Observable<News>
	func news(source: String, page: Int?) -> Observable<Resource<[News]>>
}

File: android/app/src/main/java/org/fs/news/net/repo/NewsRepository.kt
/*
 *  Copyright (C) 2019 Fatih, News Stand Android Kotlin.
 *  
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.fs.news.net.repo

import io.reactivex.Observable
import org.fs.news.model.entity.News
import org.fs.news.net.model.Resource

interface NewsRepository {

  fun bookmarks(): Observable<List<News>>
  fun hasBookmark(url: String): Observable<News>
  fun addBookmark(news: News): Observable<News>
  fun removeBookmark(news: News): Observable<News>

  fun news(sources: String, page: Int? = null): Observable<Resource<List<News>>>
}